server/test/auth.spec.js

```diff
@@ -0,0 +1,224 @@
+require("dotenv").config({ path: __dirname + "/../sample.env" });
+const chai = require("chai");
+const chaiHttp = require("chai-http");
+const sinon = require("sinon");
+const { app } = require("../app.js");
+const User = require("../models/User");
+
+chai.should();
+chai.use(chaiHttp);
+
+let agent;
+let clock;
+
+const existingUser = {
```

Comment:
is this an existing User in your mongo db? if it's a local db then this test would likely not pass for your teammates

Reply:
Unit tests [should not depend on a local running server](https://dev-tester.com/dead-simple-api-tests-with-supertest-mocha-and-chai/).

Reply:
Yes, it was in my MongoDB. So, I am modifying ```auth.spec.js``` now.
I am trying to create ```testUser``` with ```beforeEach``` and delete the ```testUser``` with ```afterEach``` for each test block.